tests/callbacks/test_rich_progress_bar.py

```diff
@@ -159,3 +160,29 @@ def configure_columns(self, trainer, pl_module):
 
     assert progress_bar.progress.columns[0] == custom_column
     assert len(progress_bar.progress.columns) == 1
+
+
+@RunIf(rich=True)
+def test_rich_progress_bar_display_every_n_epochs(tmpdir):
+
+    model = BoringModel()
+
+    with mock.patch(
+        "pytorch_lightning.callbacks.progress.rich_progress.Progress.stop", autospec=True
+    ) as mock_progress_stop:
+        progress_bar = RichProgressBar(display_every_n_epochs=2)
+        trainer = Trainer(
+            default_root_dir=tmpdir,
+            num_sanity_val_steps=0,
+            limit_train_batches=1,
+            max_epochs=6,
+            callbacks=progress_bar,
+        )
+        trainer.fit(model)
+    assert mock_progress_stop.call_count == 3
```

Comment:
should we instead check at what epoch the progress bar comes up?

Reply:
Updated the test! `reset` means continuing on the same bar.